starfish/imagestack/test/factories.py

```diff
@@ -0,0 +1,120 @@
+from collections import OrderedDict
+from typing import Sequence, Tuple
+
+import numpy as np
+import xarray as xr
+from scipy.ndimage import gaussian_filter
+
+from starfish.codebook.codebook import Codebook
+from starfish.experiment.builder import build_image, TileFetcher
+from starfish.experiment.builder.defaultproviders import OnesTile, tile_fetcher_factory
+from starfish.imagestack import physical_coordinate_calculator
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes, Coordinates, Features, PhysicalCoordinateTypes
+
+
+def synthetic_stack(
+        num_round: int = 4,
+        num_ch: int = 4,
+        num_z: int = 12,
+        tile_height: int = 50,
+        tile_width: int = 40,
+        tile_fetcher: TileFetcher = None,
+) -> ImageStack:
+    """generate a synthetic ImageStack
+
+    Returns
+    -------
+    ImageStack :
+        imagestack containing a tensor whose default shape is (2, 3, 4, 30, 20)
+        and whose default values are all 1.
+
+    """
+    if tile_fetcher is None:
+        tile_fetcher = tile_fetcher_factory(
+            OnesTile,
+            False,
+            {Axes.Y: tile_height, Axes.X: tile_width},
+        )
+
+    collection = build_image(
+        range(1),
+        range(num_round),
+        range(num_ch),
+        range(num_z),
+        tile_fetcher,
+    )
+    tileset = list(collection.all_tilesets())[0][1]
+
+    return ImageStack.from_tileset(tileset)
+
+
+def imagestack_with_coords_factory(stack_shape: OrderedDict, coords: OrderedDict) -> ImageStack:
```

Comment:
[Future] Should we work towards removing objects without coordinates? I imagine our code should all be tested for robustly dealing with and propagating coordinates.

Reply:
I do think we should probably move towards one factory that builds very realistic examples.  By putting all the factories together, I'm hoping we can start seeing the common stuff and avoid creating more. :)